test/test_action_network/test_action_network.py

```diff
@@ -348,6 +348,46 @@ def setUp(self, m):
             },
```

Comment:
You've added a number of methods with no test coverage. I realize that having test coverage is of relatively marginal value given that we have to mock the endpoints, and I don't want to create a ton of busywork for you, but I would love it if we could check that the call is what we expect, given the parameters input.

Reply:
Hi @shaunagm,
I reorganized the code and added tests for all functions added by me. 
Please let me know if it looks ok to you.

Reply:
These look great! Thank you for adding all these tests, Nir.